src/Lib.P2P.Tests/Multiplex/SubstreamTest.cs

```diff
@@ -0,0 +1,195 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Lib.P2P.Multiplex;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MultiFormats;
+
+namespace Lib.P2P.Tests.Multiplex
+{
+    [TestClass]
+    public class SubstreamTest
+    {
+        [TestMethod]
+        public void Seeking()
+        {
+            var stream = new Substream();
+            Assert.IsFalse(stream.CanSeek);
+            ExceptionAssert.Throws<NotSupportedException>(() => { stream.Seek(0, SeekOrigin.Begin); });
+            ExceptionAssert.Throws<NotSupportedException>(() => { stream.Position = 0; });
+            ExceptionAssert.Throws<NotSupportedException>(() =>
+            {
+                var _ = stream.Position;
+            });
+        }
+
+        [TestMethod]
+        public void Timeout()
+        {
+            var stream = new Substream();
+            Assert.IsFalse(stream.CanTimeout);
+            ExceptionAssert.Throws<InvalidOperationException>(() => { stream.ReadTimeout = 0; });
+            ExceptionAssert.Throws<InvalidOperationException>(() =>
+            {
+                var _ = stream.ReadTimeout;
+            });
+            ExceptionAssert.Throws<InvalidOperationException>(() => { stream.WriteTimeout = 0; });
+            ExceptionAssert.Throws<InvalidOperationException>(() =>
+            {
+                var _ = stream.WriteTimeout;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)